Move the documentation to the website and complete it

julienrf opened the pull request:
Fix #116 
Fix #133 
Fix #145
Fix #153

guy9 commented:
Thanks @julienrf , once complete please ask @annastuchlik and @tarzanek  for a review

julienrf commented:
The PR is ready for review, could you please have a look @tarzanek and @annastuchlik?

julienrf commented:
To make the review process easier, I’ve published the generated documentation here: https://julienrf.github.io/scylla-migrator/

guy9 commented:
thanks @julienrf cc @tzach

guy9 commented on docs/source/migrate-from-cassandra-or-parquet.rst:
```diff
@@ -0,0 +1,208 @@
+=============================================
+Migrate from Cassandra or from a Parquet File
```
Apache Cassandra

guy9 commented on docs/source/migrate-from-cassandra-or-parquet.rst:
```diff
@@ -0,0 +1,208 @@
+
+This page explains how to fill the ``source`` and ``target`` properties of the `configuration file </configuration>`_ to migrate data:
+
+- from Cassandra, ScyllaDB, or from a `Parquet <https://parquet.apache.org/>`_ file,
```
Apache Cassandra + everywhere else please replace Cassandra -> Apache Cassandra

guy9 commented on docs/source/migrate-from-dynamodb.rst:
```diff
@@ -0,0 +1,217 @@
+
+Where ``<access-key>`` and ``<secret-key>`` should be replaced with your actual AWS access key and secret key.
+
+The Migrator also supports advanced AWS authentication options such as using `AssumeRole <https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html>`_. Please read the `configuration reference </configuration#aws-authentication>`_ for more details.
```
Couldn't get the "configuration reference" link to work

julienrf replied:
Ah, good catch. It was working locally but it does not work on GitHub pages because the website URL now contains a prefix “/scylla-migrator”. I will fix this.

julienrf commented:
@guy9  I addressed all your comments in [eb23794](https://github.com/scylladb/scylla-migrator/pull/166/commits/eb23794c9a5e5188de68d94ea4521c1fb4f23516).

I’ve updated the website preview at https://julienrf.github.io/scylla-migrator.

tarzanek commented on docs/source/stream-changes.rst:
```diff
@@ -0,0 +1,33 @@
+
+In this mode, the migrator has to be interrupted manually with ``Control`` + ``C`` (or by sending a ``SIGINT`` signal to the ``spark-submit`` process). Currently, the created stream is not deleted when the migrator is stopped. You have to delete it manually (e.g. via the AWS Console).
+
+Note that for the migration to be performed losslessly, the initial snapshot transfer must complete within 24 hours. Otherwise, some captured changes may be lost due to the retention period of the table’s stream.
```
"losslessly" - there is no such english word afaik

"without loosing writes" (due to default stream TTL)
would that work?

julienrf replied:
Agreed, fixed in [4da9711](https://github.com/scylladb/scylla-migrator/pull/166/commits/4da971193facbee455c49626955729ec21a54a4d).

tarzanek commented on docs/source/configuration.rst:
```diff
@@ -0,0 +1,388 @@
+
+This page documents the schema of the YAML configuration file used by the Migrator and the Validator.
+
+The configuration file is a YAML object whose fields are enumerated, preceded by a comment describing their role. All the fields are mandatory except if their documentation starts with “Optional”. All the values of the form ``<xxx>`` are placeholders that should be replaced with your specific settings.
```
add something like:
YAML format is whitespace sensitive, make sure you use proper number of spaces before to keep properties on same level.


(breaking whitespace is most common user mistake, I guess this is the weakest(strongest?:-D) spot of python and yaml)

julienrf replied:
Fixed in [4da9711](https://github.com/scylladb/scylla-migrator/pull/166/commits/4da971193facbee455c49626955729ec21a54a4d).

tarzanek commented on docs/source/configuration.rst:
```diff
@@ -0,0 +1,388 @@
+  skipTokenRanges: []
+
+These top-level properties are documented in the following sections (except ``skipTokenRanges``, which is used internally).
+
```
Maybe we should provide a sample of error when something is missing or there is a typo (this DropDown ... cryptic error the config file parser prints)

Decoding that error is the second most seen user error - they don't know what to fix and where

tarzanek replied:
oh, fixed in circe 0.14.7
but let's add at least sample if something is missing, e.g.

```
io.circe.DecodingFailure$DecodingFailureImpl: DecodingFailure at .skipTokenRanges: Missing required field
```
so people can find such config errors using google

julienrf replied:
Fixed in [4da9711](https://github.com/scylladb/scylla-migrator/pull/166/commits/4da971193facbee455c49626955729ec21a54a4d).
